Sources/Instrumentation/Tracing/Span.swift

```diff
@@ -161,6 +153,39 @@ extension SpanAttribute: ExpressibleByArrayLiteral {
     }
 }
 
+/// A collection of `SpanAttribute`s.
+public struct SpanAttributes {
+    private var _attributes = [String: SpanAttribute]()
+
+    /// Accesses the `SpanAttribute` with the given name for reading and writing.
+    /// - Parameter name: The name of the attribute used to identify the attribute.
+    /// - Returns:The `SpanAttribute` identified by the given name, or `nil` if it's not present.
+    public subscript(_ name: String) -> SpanAttribute? {
+        get {
+            self._attributes[name]
+        } set {
+            self._attributes[name] = newValue
+        }
+    }
+
+    /// Calls the given callback for each attribute stored in this collection.
+    /// - Parameter callback: The function to call for each attribute.
+    public func forEachAttribute(_ callback: (String, SpanAttribute) -> Void) {
+        self._attributes.forEach { callback($0.key, $0.1) }
+    }
```

Comment:
👍 forEach would have been fine IMO, people don't have to realize they use a wrapper

Reply:
(same in other places actually, I don't think I voiced this on BaggageContext but the same thoughts tbh -- very minor thing tho)

Reply:
True, `attributes.forEachAttribute` sounds a bit repetitive. I'll change here and create an issue for it in the `BaggageContext` repository.